pkg/sql/sem/tree/overload.go

```diff
@@ -161,11 +161,12 @@ func (HomogeneousType) String() string {
 	return "anyelement..."
 }
 
-// VariadicType is a TypeList implementation which accepts any number of
-// arguments and matches when each argument is either NULL or of the type
-// typ.
+// VariadicType is a TypeList implementation which accepts a fixed number of
+// arguments at the beginning and an arbitrary number of homogenous arguments
+// at the end.
 type VariadicType struct {
-	Typ types.T
+	FixedTypes []types.T
+	Splat      types.T
```

Comment:
nit: I think this should be called VarType or left as Typ rather than Splat.

Reply:
Done.